### ISSUE TITLE
MD027: False positive on table cell pipe characters

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#268]
## Summary

MD027 (no-multiple-space-blockquote) incorrectly flags table cell padding as blockquote violations.

## Example

```markdown
| <code>&vert;</code>       | <code>pat &vert; pat</code>                             | Pattern alternatives |
```

This table row is flagged with:
```
MD027/no-multiple-space-blockquote: Multiple spaces after blockquote symbol: found 7 whitespace characters, expected 1
```

## Root Cause

The rule is detecting `|` characters followed by spaces as blockquote markers, but these are table cell delimiters, not blockquote symbols (`>`).

## Expected Behavior

MD027 should only flag lines that start with `>` (blockquote marker), not table rows with `|` (pipe characters).

## Impact

On the Rust Book corpus, we report 44 MD027 violations while rumdl reports 2 and markdownlint reports 0. Most of ours are false positives from tables.

## Related

Issue #263 - Linter comparison
